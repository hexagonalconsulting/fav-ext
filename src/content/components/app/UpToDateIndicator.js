import React from 'react'
import PropTypes from 'prop-types';
import customizedPulseLoader from './customizedPulseLoader'


const UpToDateIndicator = ({timestampsNeitherIsNull, timestampsAreEqual}) => {

  let upToDateIndicatorColor;
  // Color code meanings, upToDateIndicatorColor will either be:
  //'#00C851' => green success
  //'#ffbb33' => orange warning
  timestampsNeitherIsNull && ( upToDateIndicatorColor = { color: timestampsAreEqual ? '#00C851' : '#ffbb33' });

  const UP_TO_DATE = 'UP TO DATE';
  const NOT_UP_TO_DATE = 'NOT UP TO DATE';

  return(
    <div style={ {...{ fontWeight: 'bold', width: 150, textAlign: 'center' }, ...upToDateIndicatorColor} }>
      { !timestampsNeitherIsNull && customizedPulseLoader /* Show the customizedPulseLoader as long either of the timestamps is null. */}
      { timestampsNeitherIsNull && (
        timestampsAreEqual
          ? UP_TO_DATE
          : NOT_UP_TO_DATE
        )
      }
    </div>
  )
};

UpToDateIndicator.propTypes = {
  timestampsNeitherIsNull: PropTypes.bool.isRequired,
  timestampsAreEqual: PropTypes.bool.isRequired,
};

export default UpToDateIndicator;