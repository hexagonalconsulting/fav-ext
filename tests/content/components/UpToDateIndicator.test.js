import React from 'react'
import renderer from 'react-test-renderer';
import UpToDateIndicator from '../../../src/content/components/UpToDateIndicator'

test('shows the loader when there is a null value in the timestamps', () => {
  const component = renderer.create(
    <UpToDateIndicator
      timestampsAreEqual={false}
      timestampsNeitherIsNull={false}
    />
  );
  expect(component).toMatchSnapshot()
});

test('shows the loader when both timestamps are null', () => {
  const component = renderer.create(
    <UpToDateIndicator
      timestampsAreEqual={true}
      timestampsNeitherIsNull={false}
    />
  );
  expect(component).toMatchSnapshot()
});

test('shows the tab is "UP TO DATE" if the timestamps are equal', () => {
  const component = renderer.create(
    <UpToDateIndicator
      timestampsAreEqual={true}
      timestampsNeitherIsNull={true}
    />
  );
  expect(component).toMatchSnapshot()
});

test('shows the tab is "NOT UP TO DATE" if the timestamps are not equal', () => {
  const component = renderer.create(
    <UpToDateIndicator
      timestampsAreEqual={false}
      timestampsNeitherIsNull={true}
    />
  );
  expect(component).toMatchSnapshot()
});