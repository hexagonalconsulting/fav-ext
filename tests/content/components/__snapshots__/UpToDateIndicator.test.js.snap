// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows the loader when both timestamps are null 1`] = `
<div
  style={
    Object {
      "fontWeight": "bold",
      "textAlign": "center",
      "width": 150,
    }
  }
>
  <div
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <div>
      <div
        className="css-msggfa"
      />
      <div
        className="css-dyhi6e"
      />
      <div
        className="css-1q065sq"
      />
    </div>
  </div>
</div>
`;

exports[`shows the loader when there is a null value in the timestamps 1`] = `
<div
  style={
    Object {
      "fontWeight": "bold",
      "textAlign": "center",
      "width": 150,
    }
  }
>
  <div
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <div>
      <div
        className="css-msggfa"
      />
      <div
        className="css-dyhi6e"
      />
      <div
        className="css-1q065sq"
      />
    </div>
  </div>
</div>
`;

exports[`shows the tab is "NOT UP TO DATE" if the timestamps are not equal 1`] = `
<div
  style={
    Object {
      "color": "#ffbb33",
      "fontWeight": "bold",
      "textAlign": "center",
      "width": 150,
    }
  }
>
  NOT UP TO DATE
</div>
`;

exports[`shows the tab is "UP TO DATE" if the timestamps are equal 1`] = `
<div
  style={
    Object {
      "color": "#00C851",
      "fontWeight": "bold",
      "textAlign": "center",
      "width": 150,
    }
  }
>
  UP TO DATE
</div>
`;
